addons/account/models/partner.py

```diff
@@ -434,6 +434,17 @@ def _compute_bank_count(self):
         for partner in self:
             partner.bank_account_count = mapped_data.get(partner.id, 0)
 
+    @api.constrains('company_id')
+    def _check_company_and_amls(self):
+        amls = self.env['account.move.line'].sudo().search([
+            ('partner_id', 'in', self.ids),
+            ('reconciled', '=', False),
+            ('account_id.deprecated', '=', False),
+            ('move_id.state', '=', 'posted'),
+        ], limit=1)
```

Comment:
The limit like it is here is just wrong.
We have 3 solutions:
* The one you have before, without the limit
* Doing a search in a loop
```python
for partner in self:
    if self.env['account.move.line'].sudo().search([
        ('partner_id', '=', partner.id),
        ('company_id', '!=', partner.company_id.id),
        ('reconciled', '=', False),
        ('account_id.deprecated', '=', False),
        ('move_id.state', '=', 'posted'),
    ], limit=1):
        raise blablabla
```
* Best one for the perfs: searching for it with a SQL query.

I tried using a `read_group` but I couldn't find a working combination.